### PR TITLE
feat: Add version-based resource type compatibility checks 

### PR DESCRIPTION
**Pull request by RovinKYK**
## Purpose                                                                                                                                                                            
                                                                                                                                                                                        
The iamctl tool previously had no mechanism to gate resource type operations based on the Identity Server version configured by the user. This meant that attempting to export or import resource types unsupported by the target IS version would result in API errors, with no clear feedback to the user.  

Related to https://github.com/wso2-enterprise/iam-product-management/issues/662 
Merge After: https://github.com/wso2-extensions/identity-tools-cli/pull/53                            
                                                                                                                                                                                        
  ## Goals                                                  

  - Introduce minimum version enforcement per resource type, skipping unsupported operations with a clear log message when the configured server version is below the required threshold.
  - Introduce maximum version enforcement per resource type, allowing future deprecation of resource types that are no longer supported beyond a certain IS version.
  - Support both 2-digit (`MAJOR.MINOR`) and 3-digit (`MAJOR.MINOR.PATCH`) version strings for flexibility.

  ## Approach

  Added two new files under `pkg/utils/`:

  - **`versionRequirements.go`** — Defines `EntityMinVersionRequirements` and `EntityMaxSupportedVersion` maps keyed by `ResourceType`. These act as a central registry for version bounds. Adding support bounds for a new resource type only requires updating these maps.
  - **`versionUtils.go`** — Implements `IsEntitySupportedInVersion(resourceType)`, which reads the configured `ServerVersion` and checks it against both the min and max bounds for the given resource type. Returns `true` (allowing the operation) if no version is configured (backward compatibility), no bounds are defined for the type, or the version falls within the defined range. Logs a descriptive skip message otherwise.

  `ParseVersion` and `CompareVersions` helpers handle semantic version parsing and support both 2-digit and 3-digit formats. Resource packages can call `IsEntitySupportedInVersion` at the start of its `ExportAll`/`ImportAll` functions.

  A full test suite in `tests/versionUtils_test.go` covers version comparison, parsing edge cases, and all boundary conditions for the min/max version checks.

## User stories
> Summary of user stories addressed by this change>

## Release note
> Brief description of the new feature or bug fix as it will appear in the release notes

## Documentation
> Link(s) to product documentation that addresses the changes of this PR. If no doc impact, enter “N/A” plus brief explanation of why there’s no doc impact

## Training
> Link to the PR for changes to the training content in https://github.com/wso2/WSO2-Training, if applicable

## Certification
> Type “Sent” when you have provided new/updated certification questions, plus four answers for each question (correct answer highlighted in bold), based on this change. Certification questions/answers should be sent to certification@wso2.com and NOT pasted in this PR. If there is no impact on certification exams, type “N/A” and explain why.

## Marketing
> Link to drafts of marketing content that will describe and promote this feature, including product page changes, technical articles, blog posts, videos, etc., if applicable

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

## Samples
> Provide high-level details about the samples related to this feature

## Related PRs
> List any other related PRs

## Migrations (if applicable)
> Describe migration steps and platforms on which migration has been tested

## Test environment
> List all JDK versions, operating systems, databases, and browser/versions on which this feature/fix was tested
 
## Learning
> Describe the research phase and any blog posts, patterns, libraries, or add-ons you used to solve the problem.

### CODE REVIEW

File: iamctl/pkg/governanceConnectors/export.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+	"github.com/wso2-extensions/identity-tools-cli/iamctl/pkg/utils"
+)
+
+func ExportAll(exportFilePath string, format string) {
+
+	log.Println("Exporting governance connectors...")
+	exportFilePath = filepath.Join(exportFilePath, utils.GOVERNANCE_CONNECTORS.String())
+
+	if utils.IsResourceTypeExcluded(utils.GOVERNANCE_CONNECTORS) {
```
Comment (HasiniSama) [inline iamctl/pkg/governanceConnectors/export.go:35]: Is it intentional that there’s no **IsEntitySupportedInVersion** check for governance connectors?

File: iamctl/pkg/utils/constants.go [hunk 2]
```diff
@@ -181,10 +186,16 @@ var challengeQuestionsArrayFields = []string{
 	"questions",
 }
 
+var governanceConnectorArrayFields = []string{
+	"properties",
+}
```
Comment (HasiniSama) [inline iamctl/pkg/utils/constants.go:191]: Where is **governanceConnectorArrayFields** used?

File: iamctl/pkg/scriptLibraries/import.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+	existingList, err := getScriptLibraryList()
+	if err != nil {
+		log.Println("Error retrieving the deployed script library list: ", err)
+		return
+	}
+
+	files, err := os.ReadDir(importFilePath)
+	if err != nil {
+		log.Println("Error importing script libraries: ", err)
```
Comment (HasiniSama) [inline iamctl/pkg/scriptLibraries/import.go:51]: In other places we return after an error, why do we continue execution here?

File: iamctl/pkg/applications/export.go [hunk 4]
```diff
@@ -35,6 +35,9 @@ func ExportAll(exportFilePath string, format string) {
 	log.Println("Exporting applications...")
 	exportFilePath = filepath.Join(exportFilePath, utils.APPLICATIONS.String())
 
+	if !utils.IsEntitySupportedInVersion(utils.APPLICATIONS) {
+		return
+	}
```
Comment (HasiniSama) [inline iamctl/pkg/applications/export.go:40]: Do we need to check **IsEntitySupportedInVersion** for each of the core resource types (applications, claims, identity providers, user stores)?

If a given IS version doesn’t support them, won’t the rest also be unsupported?

File: iamctl/pkg/scriptLibraries/import.go (hunk 3 above)
Comment (HasiniSama) [reply iamctl/pkg/scriptLibraries/import.go:51]: We should return here, otherwise following logic will throw and error.

File: iamctl/pkg/governanceConnectors/export.go (hunk 1 above)
Comment (RovinKYK) [reply iamctl/pkg/governanceConnectors/export.go:35]: Yes, currently governance connectors are supported in all IS versions where IAM CTL is supported.

File: iamctl/pkg/utils/constants.go (hunk 2 above)
Comment (RovinKYK) [reply iamctl/pkg/utils/constants.go:191]: It is used in `GetArrayFieldPaths` function in serializationUtils

File: iamctl/pkg/applications/export.go (hunk 4 above)
Comment (RovinKYK) [reply iamctl/pkg/applications/export.go:40]: Still the CRUD Api based import/export is not implemented for these core resources. That is the reason for having this version gate. Later this is removed after implementing the CRUD Api based path in a later PR.

**Review by lashinijay — DISMISSED** [review]

**Review by lashinijay — APPROVED** [review]
